src/main/kotlin/app/revanced/patches/twitter/misc/links/ChangeLinkSharingDomainPatch.kt

```diff
@@ -0,0 +1,140 @@
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchException
+import app.revanced.patcher.patch.annotation.CompatiblePackage
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patcher.patch.options.PatchOption.PatchExtensions.stringPatchOption
+import app.revanced.patches.twitter.misc.links.fingerprints.LinkBuilderFingerprint
+import app.revanced.patches.twitter.misc.links.fingerprints.LinkResourceGetterFingerprint
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.BuilderInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.TwoRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+
+
+@Patch(
+    name = "Change link sharing domain",
+    description = "Replaces the domain name of Twitter links when sharing them.",
+    compatiblePackages = [CompatiblePackage("com.twitter.android")]
+)
+@Suppress("unused")
+object ChangeLinkSharingDomainPatch : BytecodePatch(
+    setOf(LinkBuilderFingerprint, LinkResourceGetterFingerprint)
+) {
+    private var domainName by stringPatchOption(
+        key = "domainName",
+        default = "fxtwitter.com",
+        title = "Domain name",
+        description = "The domain to use when sharing links.",
+        required = true
+    )
+
+    override fun execute(context: BytecodeContext) {
+        val linkBuilderResult = LinkBuilderFingerprint.result ?: throw LinkBuilderFingerprint.exception
+
+        // region Copy link button.
+        linkBuilderResult.mutableMethod.apply {
+            val stringIndex = linkBuilderResult.scanResult.stringsScanResult!!.matches
+                .first().index
+
+            val targetRegister = getInstruction<OneRegisterInstruction>(stringIndex).registerA
+            replaceInstruction(
+                stringIndex,
+                "const-string v$targetRegister, \"https://$domainName/%1\$s/status/%2\$d\""
+            )
+        }
+
+
+        // endregion
+
+        // Used in the Share via... dialog.
+        LinkResourceGetterFingerprint.result?.apply {
+            val instructions = mutableMethod.getInstructions()
+
+            // Result register of the original method call.
+            var resultRegister = 0
+
+            // Remove instructions at the end and get the register of the last instruction.
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Landroid/content/res/Resources;") continue
```

Comment:
What is the strategy here? Why do you need to loop over all methods?

Reply:
Here we find the og method call to remove, I guess I should replace these with find or something cause this for is just used to find the method